test/integration/imagechange_buildtrigger_test.go

```diff
@@ -477,12 +479,21 @@ func TestMultipleImageChangeBuildTriggers(t *testing.T) {
 		if err != nil {
 			t.Fatalf("Couldn't create Image: %v", err)
 		}
+
+		var newBuild *buildapi.Build
+		var event watchapi.Event
 		// wait for initial build event from the creation of the imagerepo
-		event := <-watch.ResultChan()
+		for {
+			event = <-watch.ResultChan()
+			newBuild = event.Object.(*buildapi.Build)
```

Comment:
This can panic. There's no guarantee the object is a `*buildapi.Build`, it can be an error event in a `*unversioned.Status` (as I saw in #7174).
We could at least check it with the double assignment `_, ok = event.Object.(*buildapi.Build)` syntax, and `t.Errorf` if `!ok`.